runtime/near-vm-logic/src/dependencies.rs

```diff
@@ -486,4 +531,17 @@ pub trait External {
 
     /// Returns total stake of validators in the current epoch.
     fn validator_total_stake(&self) -> Result<Balance>;
+
+    /// Distribute the gas among the scheduled function calls that specify a gas weight.
+    /// Returns the amount of distributed gas.
+    ///
+    /// # Arguments
+    ///
+    /// * `gas` - amount of unused gas to distribute
+    ///
+    /// # Returns
+    ///
+    /// Function returns true if gas was distributed, false if no ratios existed to distribute to.
```

Comment:
Ah, it would be an outdated comment above, I can update that.

I'm a little skeptical to add an additional type to the public trait API, are you sure this is in line with all usages? Seems as if most of the API is primitive types

Reply:
My concern is largely rooted in the fact that this function mutates some state in a semi-fallible looking manner and makes its outcome easy to ignore, or handle incorrectly. There seems to be little in the way to prevent somebody from forgetting to deal with the gas that has been distributed by the call to this function. Not to mention a `false` return is not really distinguishable between “this is called twice, neard is buggy” and “there is no calls with weights”.

Are there use-cases where distributing the gas but not prepaying it is useful? Maybe prepaying inside the implementation of this method would make sense?

> Seems as if most of the API is primitive types

What do you mean? I believe use of primitive types in signatures is only required for VM host function exports, but this isn't such a function.

Reply:
yeah, this is fair. I think I'll go with a `must_use` enum in this case, since having no ratios isn't actually an "error".

> What do you mean? I believe use of primitive types in signatures is only required for VM host function exports, but this isn't such a function.

I was just noting that it's not ergonomic to have a bunch of custom types as part of the trait API, requiring implementations to import the type and requiring importing this type when calling the function.